wsd-frontend/src/components/wsd/MemePost/MemePost.tsx

```diff
@@ -0,0 +1,155 @@
+import type { Metadata } from 'next'
+import Link from 'next/link'
+import { notFound } from 'next/navigation'
+
+import _ from 'lodash'
+
+import { Button, buttonVariants } from '@/components/shadcn/button'
+import { Overlay, OverlayClose, OverlayContent, OverlayTitle, OverlayTrigger } from '@/components/shadcn/overlay'
+import { Separator } from '@/components/shadcn/separator'
+import AuthenticatedOnlyActionButton from '@/components/wsd/AuthenticatedOnlyActionButton'
+import Meme from '@/components/wsd/Meme'
+import MemeComment from '@/components/wsd/MemeComment'
+import NewComment from '@/components/wsd/NewComment'
+
+import { APIQuery, APIType, includesType } from '@/api'
+import config from '@/config'
+import { getWSDMetadata } from '@/lib/metadata'
+import { useWSDAPI as sUseWSDAPI } from '@/lib/serverHooks'
+import { getKeys } from '@/lib/typeHelpers'
+import { InvalidHEXError, cn, hexToUUIDv4, suppress } from '@/lib/utils'
+
+export async function generatePostAndCommentMetadata(props: {
+  params: Promise<{ hex: string; commentHex?: string }>
+}): Promise<Metadata | undefined> {
+  const params = await props.params
+  const wsd = sUseWSDAPI()
+  const postId = suppress<string, undefined>([InvalidHEXError], () => hexToUUIDv4(params.hex))
+  const commentParamId = params.commentHex
+
+  if (!_.isUndefined(postId)) {
+    const { data: post } = await wsd.post(postId, { include: 'tags' })
+    if (!_.isUndefined(post)) {
+      if (!_.isUndefined(commentParamId)) {
+        const commentId = suppress<string, undefined>([InvalidHEXError], () => hexToUUIDv4(commentParamId))
+        if (!_.isUndefined(commentId)) {
+          const { data: comment } = await wsd.postComment(commentId, { include: 'user' })
+          if (!_.isUndefined(comment)) {
+            const commentData = includesType(comment, 'user', 'User')
+
+            // @TODO: Handle comment body
```

Comment:
Are you looking to render the comment and then return the text? That might be slightly complicated, we can render the comments on the backend but we can't get the text easily, we can throw in something like jsdom to emulate dom and use some dom method to get the text, that's the only method I can think of right now.

Reply:
Yes but i dont know how to do it really, i would rather butcher your fine code even more.